smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestResponseGenerator.kt

```diff
@@ -55,23 +60,63 @@ open class HttpProtocolUnitTestResponseGenerator protected constructor(builder:
     }
 
     open fun renderExpectedBody(test: HttpResponseTestCase) {
-        if (test.body.isPresent && test.body.get().isNotBlank()) {
+        if (test.body.isPresent) {
             operation.output.ifPresent {
-                val outputShape = model.expectShape(it) as StructureShape
-                val data = writer.format("Data(\"\"\"\n\$L\n\"\"\".utf8)", test.body.get().replace("\\\"", "\\\\\""))
-                // depending on the shape of the output, we may need to wrap the body in a stream
-                if (outputShape.hasStreamingMember(model)) {
-                    // wrapping to CachingStream required for test asserts which reads body multiple times
-                    writer.write("content: .stream(\$N(data: \$L, isClosed: true))", SmithyStreamsTypes.Core.BufferedStream, data)
+                if (test.body.get().isNotBlank()) {
+                    val outputShape = model.expectShape(it) as StructureShape
+                    val data = writer.format(
+                        "Data(\"\"\"\n\$L\n\"\"\".utf8)",
+                        test.body.get().replace("\\\"", "\\\\\"")
+                    )
+                    // depending on the shape of the output, we may need to wrap the body in a stream
+                    if (outputShape.hasStreamingMember(model)) {
+                        // wrapping to CachingStream required for test asserts which reads body multiple times
+                        writer.write(
+                            "content: .stream(\$N(data: \$L, isClosed: true))",
+                            SmithyStreamsTypes.Core.BufferedStream,
+                            data
+                        )
+                    } else {
+                        writer.write("content: .data(\$L)", data)
+                    }
+                } else if (test.body.get().isBlank() && bodyHasDefaultValue()) {
+                    // Expected body is blank but not because it's nil, but because it's a default empty blob value
+                    writer.write("content: .data(Data(\"\".utf8))")
                 } else {
-                    writer.write("content: .data(\$L)", data)
+                    // Expected body is blank and underlying member shape does not have default values.
+                    writer.write("content: nil")
```

Comment:
Before we used to equate empty `body` string in the Smithy protocol test as `nil`. But an empty `body` can mean an empty blob of size 0 if one of the body members has a default value of an empty blob. Change here adds handling that edge case.